pvlib/irradiance.py

```diff
@@ -1212,6 +1225,235 @@ def perez(surface_tilt, surface_azimuth, dhi, dni, dni_extra,
         return sky_diffuse
 
 
+def _calc_delta(dhi, dni_extra, solar_zenith, airmass=None):
+    '''
+    Compute the delta parameter, which represents sky dome "brightness"
+    in the Perez and Perez-Driesse models.
+
+    Helper function for perez_driesse transposition.
+    '''
+    if airmass is None:
+        # use the same airmass model as in the original perez work
+        airmass = get_relative_airmass(solar_zenith, 'kastenyoung1989')
+
+    max_airmass = get_relative_airmass(90, 'kastenyoung1989')
+    airmass = np.where(solar_zenith >= 90, max_airmass, airmass)
+
+    return dhi / (dni_extra / airmass)
+
+
+def _calc_zeta(dhi, dni, zenith, use_kappa=True):
+    '''
+    Compute the zeta parameter, which represents sky dome "clearness"
+    in the Perez-Driesse model.
+
+    Helper function for perez_driesse transposition.
+    '''
+    dhi = np.asarray(dhi)
+    dni = np.asarray(dni)
+
+    with np.errstate(invalid='ignore'):
+        zeta = dni / (dhi + dni)
+
+    zeta = np.where(dhi == 0, 0.0, zeta)
+
+    if use_kappa:
```

Comment:
I guess `use_kappa=False` switches back to the 1987 model?  Unless I missed it, the reference makes no mention of the kappa term being optional in the Perez-Driesse model.  I suggest we remove the `use_kappa` parameter and make the calculations in this function structured more similarly to the equations in the reference.

Reply:
The option is left over from development and I thought it might be interesting for other experimenters.  I can take it out if you want. Not sure what you mean by the last bit though.

Reply:
The last bit meant that it took me a few moments to mentally rework Eqs 6, 7, and 13 in the reference to match the calculations being done here.  Not a big deal :)

If you'd rather keep the code the way it is now, maybe a brief explanatory comment would save future readers the minute or two it took me to convince myself I understood this function.

Reply:
I will simplify this code but keep the direct calculation.